### PR TITLE
Fix mui-datatable's options: onRowsDelete() callback signature

### PR DESCRIPTION
**Pull request by francoisprunier**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/gregnb/mui-datatables/blob/6d448052ae2cd73819efcd76ab67b9c9e407429d/src/MUIDataTable.js#L1081
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mui-datatables/index.d.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ export interface MUIDataTableOptions {
     onFilterDialogOpen?: () => void;
     onFilterDialogClose?: () => void;
     onRowClick?: (rowData: string[], rowMeta: { dataIndex: number; rowIndex: number }) => void;
-    onRowsDelete?: (rowsDeleted: any[]) => void;
+    onRowsDelete?: (rowsDeleted: {lookup: {[dataIndex: number]: boolean}, data: Array<{index: number, dataIndex: number}>}) => void;
```
Comment (Favna) [inline types/mui-datatables/index.d.ts:233]:
```suggestion
onRowsDelete?: (rowsDeleted: {
        lookup: { [dataIndex: number]: boolean };
        data: Array<{ index: number; dataIndex: number }>;
    }) => void;
```

Applied some autoformatter to make this much more readable

File: types/mui-datatables/mui-datatables-tests.tsx [hunk 2]
```diff
@@ -99,6 +99,11 @@ const MuiCustomTable: React.FC<Props> = (props) => {
                     });
             }
         },
+        onRowsDelete: (rowsDeleted: {lookup: {[dataIndex: number]: boolean}, data: Array<{index: number, dataIndex: number}>}) => {
+            if (rowsDeleted.data[0].index === rowsDeleted.data[0].dataIndex && rowsDeleted.lookup[0]) {
+                console.log(`Data deleted on index ${rowsDeleted.data[0].dataIndex}`);
+            }
+        },
```
Comment (Favna) [inline types/mui-datatables/mui-datatables-tests.tsx:106]:
```suggestion
        onRowsDelete: (rowsDeleted: {
            lookup: { [dataIndex: number]: boolean };
            data: Array<{ index: number; dataIndex: number }>;
        }) => {
            if (rowsDeleted.data[0].index === rowsDeleted.data[0].dataIndex && rowsDeleted.lookup[0]) {
                console.log(`Data deleted on index ${rowsDeleted.data[0].dataIndex}`);
            }
        },
```
Also formatting

**Review by favna — CHANGES_REQUESTED** [review]
Once you resolve Travis issues (note: you can run the same test suite locally with `npm run test`!) and cover these 2 changes I'll approve.

**Comment by francoisprunier** [comment]
> Once you resolve Travis issues (note: you can run the same test suite locally with `npm run test`!) and cover these 2 changes I'll approve.

Thanks, I did run `npm run test` successfully on my machine before submitting the PR, not sure why it's not behaving in the same way. I'll check.

**Comment by francoisprunier** [comment]
Hey @Favna,

I've had a look and I don't see why the CI build is failing: it's complaining that
`  All declarations in this module are exported automatically.`, however I haven't changed the exports and `npm run test` is fine on my machine.

Do you have an idea about what's happening here ?

**Review by favna — APPROVED** [review]

**Comment by Favna** [comment]
@francoischalifour DT probably changed the rules of dtslint, which is why this is triggering. Do one of the following:
1. Ensure there is an `export` for every interface / type / etc  (preferred)
2. Add the mentioned `export {}` in the file (not preferred)

**Comment by byrekt** [comment]
Hm...looks like they did update the CI package 2 days ago...seems like other modules are running into issues as well. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45101

**Comment by francoisprunier** [comment]
> Hm...looks like they did update the CI package 2 days ago...seems like other modules are running into issues as well. #45101

The weird thing is that I've got this config as well since I forked earlier today. And it was passing on my machine.

**Comment by byrekt** [comment]
@francoisprunier yeah I just pulled it down as well. Same for me. That's not good. It's going to frustrate a lot of people if they CI check doesn't run the same locally. 

```1> mui-datatables failing:
1> Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/mui-datatables/index.d.ts:465:1
1> ERROR: 465:1  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md```

**Comment by byrekt** [comment]
Looks like it now just has an issue with the default export at the end of the file.

**Comment by francoisprunier** [comment]
> @francoisprunier yeah I just pulled it down as well. Same for me. That's not good. It's going to frustrate a lot of people if they CI check doesn't run the same locally.
> 
> ```
> 1> Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/mui-datatables/index.d.ts:465:1
> 1> ERROR: 465:1  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md```
> ```

I am frustrated indeed !

> Looks like it now just has an issue with the default export at the end of the file.

Yup, however I don't know how to fix that. If I remove it, it complains that I don't have it anymore... Looks like it's related to typescript 4.0 however, maybe there's a way to disable the check for that version ?

**Review by favna — APPROVED** [review]
I have no idea what went down, Travis ran again, @TypeScriptBot mentioned me about new commits but GitHub shows nothing. Regardless it looks like the CI passed so here's my approval for a merge before the CI messes up again

p.s. finally no more travis in the future, see 44906 :D !

**Comment by sandersn** [comment]
@Favna it's probably because I manually re-ran the failed Travis run and the bot wasn't expecting it. (I will be happy to see Travis go too. :)

**Comment by francoisprunier** [comment]
Ready to merge

**Comment by francoisprunier** [comment]
@sandersn @Favna @byrekt Thanks guys, we got it merged!. I definitely wasn't expecting so much trouble from CI :)

**Pull request merged**